File: activity/20200805/RabbitLee/Self_Dividing_Numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        ans = list()
        for num in range(left, right + 1):
            if self.isSelfDividingNumber(num):
                ans.append(num)
        return ans
        
    def isSelfDividingNumber(self, num: int) -> bool:
        num2 = num
        while num2 > 0:
            digit = num2 % 10
            if digit == 0 or num % digit != 0:
                return False
            num2 //= 10
        else:
            return True

File: activity/20200805/scau-flyingbird/Self_Dividing_Numbers.go
func isSelfDividing(num int) bool {
	tmp := num
	for tmp != 0 {
		m := tmp % 10
		if m == 0 {
			return false
		}
		if num%m != 0 {
			return false
		}
		tmp /= 10
	}
	return tmp == 0
}

func selfDividingNumbers(left int, right int) []int {
	num := []int{}
	for i := left; i <= right; i++ {
		if isSelfDividing(i) {
			num = append(num, i)
		}
	}
	return num
}
